LBHackney-IT/lbh-fss-portal-frontend
# Feature/edit organisation

**Description**

Allow users to edit organisation information 

**Test instruction**

pull this branch
navigate to mock-api/account/GET.js and replace with the following content (this will ensure you are logged in as a VCSO user)
```
var mockUsers = require("../mockUsers.json");
var sample = require("lodash/sample");

module.exports = (req, res) => {
  res
    .status(200)
    .json(sample(mockUsers.filter((u) => u.name === "Tommie Dietrich")));
};
```
- navigate to **Your organisation** from the main menu, you should be redirects to **/organisations/9999/edit**
- check you can click through each of the pages using the nav bar
- check the pre-populated values in all fields are consistent with the raw data - raw data can be found in mock-api/mockOrganisations.json looking at the first organisation id: 9999, name: "Koepp Inc"
- check that if you make changes to values on a given page, click to a different page using the nav bar, then click back to the page using the nav bar these changes were not saved 
- check that if you make changes to values on a given page, click continue button at bottom of page, then click back to the page using the nav bar changes were saved (also please check that if a question has hidden fields associated with it, if you select a value which hides these fields, these fields are empty
e.g. on page 2 the value for 'Registered with the Tenant and Resident Association' is  yes, because it is yes there is an additional field visible with data in it, if you click no this field is not visible, if you click continue, then click back to this page, if you then click yes on 'Registered with the Tenant and Resident Association' the data previously in this hidden field is removed
- after clicking submit on final page (page 4) you should see 'organisation has been updated' message, if you update the mock api so the organisation has 'rejected' status you should instead see 'organisation has been submitted for review', in order to update mock-api go to **mock-api/organisations/__organisationId__/PATCH.js** and change `return item.status !== "rejected";` to `return item.status === "rejected";`


**Ticket**
https://app.clickup.com/t/9e5u8a
